ding/rl_utils/ppo.py

```diff
@@ -46,7 +52,8 @@ def ppo_error(
         data: namedtuple,
         clip_ratio: float = 0.2,
         use_value_clip: bool = True,
-        dual_clip: Optional[float] = None
+        dual_clip: Optional[float] = None,
+        happo_factor: bool = False
```

Comment:
It's better to define a new function named "shape_fn_happo", rather than combining two functions into one, which will make it difficult for maintenance. It will be confusing for RL beginers if they only want to know codes of ppo but having no idea about what happo is.